app/src/main/java/com/blockeq/stellarwallet/activities/MnemonicActivity.kt

```diff
@@ -144,5 +150,11 @@ class MnemonicActivity : BaseActivity(), View.OnClickListener {
     private fun getMnemonicList(mnemonic : String) : List<String> {
         return mnemonic.split(" ".toRegex()).dropLastWhile { it.isEmpty() }
     }
+
+    private fun generateQRCode(data: String, imageView: ImageView, size: Int) {
+        val barcodeEncoder = BarcodeEncoder()
+        val bitmap = barcodeEncoder.encodeBitmap(data, BarcodeFormat.QR_CODE, size, size)
```

Comment:
looks like you are making the UI thread to do heavy work here

Reply:
It's the same code as displaying in receive?
You want to move to background thread and then display once it is ready? :P